src/PointwiseFunctions/AnalyticSolutions/Errors.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Variables.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Tags {
+/*!
+ * \brief Prefix indicating the difference between two values, such as betwen a
+ * numeric quantity and the corresponding quantity computed from an analytic
+ * solution.
+ */
+template <typename Tag>
+struct Difference : db::PrefixTag, db::SimpleTag {
+  using type = db::const_item_type<Tag>;
+  using tag = Tag;
+};
+
+/*!
+ * \brief For each `Tag` in `TagsList`, where `TagsList` is a subset of
+ * `AnalyticTagsList`, compute the difference `Tag` - `Tags::Analytic<Tag>`.
+ */
```

Comment:
is the compiler error comprehensible if `TagsList` is not a subset of `AnalyticTagsList` ?

Reply:
I think so. If you remove WrappedGr from the solution in the test (so SpacetimeMetric, Phi, and Pi, the three tags in TagsList, are not in the analytic solution), then you get this error, which says "can't get SpacetimeMetric from Variables" and points to the precise line of Errors.hpp where you try to get that tag out of the analytic solution

error: static_assert failed due to requirement 'tmpl::list_contains_v<brigand::list<Tags::Analytic<gr::Tags::Lapse<DataVector> >, Tags::Analytic<Tags::dt<gr::Tags::Lapse<DataVector> > >, Tags::Analytic<Tags::deriv<gr::Tags::Lapse<DataVector>, brigand::integral_constant<unsigned long, 1>, Frame::Inertial, nullptr_t> >, Tags::Analytic<gr::Tags::Shift<1, Frame::Inertial, DataVector> >, Tags::Analytic<Tags::dt<gr::Tags::Shift<1, Frame::Inertial, DataVector> > >, Tags::Analytic<Tags::deriv<gr::Tags::Shift<1, Frame::Inertial, DataVector>, brigand::integral_constant<unsigned long, 1>, Frame::Inertial, nullptr_t> >, Tags::Analytic<gr::Tags::SpatialMetric<1, Frame::Inertial, DataVector> >, Tags::Analytic<Tags::dt<gr::Tags::SpatialMetric<1, Frame::Inertial, DataVector> > >, Tags::Analytic<Tags::deriv<gr::Tags::SpatialMetric<1, Frame::Inertial, DataVector>, brigand::integral_constant<unsigned long, 1>, Frame::Inertial, nullptr_t> >, Tags::Analytic<gr::Tags::SqrtDetSpatialMetric<DataVector> >, Tags::Analytic<gr::Tags::ExtrinsicCurvature<1, Frame::Inertial, DataVector> >, Tags::Analytic<gr::Tags::InverseSpatialMetric<1, Frame::Inertial, DataVector> > >, Tags::Analytic<gr::Tags::SpacetimeMetric<1, Frame::Inertial, DataVector> > >' "Could not retrieve Tag from Variables. See the first template parameter of the instantiation for what Tag is being retrieved and the second template parameter for what Tags are available." 
static_assert(tmpl::list_contains_v<TagList, Tag>,
  ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/geoffrey/Codes/spectre/spectre/src/PointwiseFunctions/AnalyticSolutions/Errors.hpp:46:14: note: in instantiation of function template

It specifically says it can't retrieve SpacetimeMetric from the Variables. The error then points to the precise line where you try to get it